packages/@aws-cdk/core/lib/lazy.ts

```diff
@@ -123,10 +123,16 @@ abstract class LazyBase implements IResolvable {
   public readonly creationStack: string[];
 
   constructor() {
-    this.creationStack = captureStackTrace();
+    // Stack trace capture is condition to CDK_DEBUG=true, because lazies can be
+    // created in a fairly thrashy way, and the stack traces are large and slow
+    // to obtain; but are mostly useful only when debugging a resolution issue.
+    this.creationStack = process.env.CDK_DEBUG === 'true'
```

Comment:
Why not any truthy value? This is different from the way we treat other boolean envvars.

Reply:
I personally think the way we do the other boolean  env vars is **wrong**: `'false'` is a truthy value, so `CDK_DEBUG=false` actually *ENABLES* debug mode, which is **ridiculous**.